### ISSUE TITLE
macOS project template should support Cmd+W shortcut to close the window

### ISSUE THREAD

**Issue by mgenware** [flutter/flutter#102439]
## Proposal

`Cmd + W` is my most used keyboard shortcut on macOS. I believe most macOS users also do that. Project created with `flutter create` currently doesn't support `Cmd + W`.

Here's a screenshot of an Electron app which has `Cmd + W` support.
<img width="188" alt="image" src="https://user-images.githubusercontent.com/11565096/164894185-f18aa565-8701-482c-91ac-dec2b8de8d09.png">

**Comment by darshankawar** [comment]
@mgenware 
Are you proposing to have the said keyboard shortcut to close the flutter app ? Currently, if we use cmd + q, it closes the app.

**Comment by mgenware** [comment]
@darshankawar 
Yes. I actually use Cmd+W way more often than Cmd+Q. It's supported on all apps I've been using including system ones.

**Comment by darshankawar** [comment]
Thanks for the update. Treating this as a proposal and for further insights from the team.

**Comment by christopherfujino** [comment]
@stuartmorgan is this really a tool feature?

**Comment by stuartmorgan** [comment]
Yes, this would be entirely within the `flutter create` template.

**Comment by stuartmorgan** [comment]
@gspencergoog What's the current state of how the menu API interacts with existing native menus? IIRC the reason I removed the File menu from the template originally is that with `menubar` it was much easier to add an entirely new menu from Dart than to merge Dart and native menus, so by not including it, it was much easier to just control the entire File menu from Dart.

Is that the case with the framework version, or not?

**Comment by anayw2001** [comment]
@gspencergoog Would https://github.com/flutter/flutter/pull/157947 be fine then or would you prefer something based in Dart on the `PlatformMenuBar` implementation?

**Comment by bkonyi** [comment]
Friendly ping @gspencergoog
